pkg/istio/rbac/common/util.go

```diff
@@ -69,3 +272,151 @@ func NewConfig(configType string, namespace string, name string, spec proto.Mess
+			if checkFn != nil && checkFn(existingConfig) {
+				continue
+			}
+			// copy metadata(for resource version) from current config to desired config
+			desiredConfig.ConfigMeta = existingConfig.ConfigMeta
+			item := Item{
+				Operation:       model.EventUpdate,
+				Resource:        desiredConfig,
+				CallbackHandler: cbHandler,
+			}
+			changeList = append(changeList, &item)
+			continue
+		}
+		// case 3: current CR is not empty, desired CR is not empty, current CR == desired CR, results in no action
+	}
+
+	// loop through the current slice of model.Config and add the items that needs to be deleted
+	for _, currConfig := range currentCRs {
+		key := currConfig.Key()
+		_, exists := desiredMap[key]
+		if checkFn != nil && checkFn(currConfig) {
+			continue
+		}
+		// case 4: current CR is not empty, desired CR is empty, results in resource deletion
+		if !exists {
+			item := Item{
+				Operation:       model.EventDelete,
+				Resource:        currConfig,
+				CallbackHandler: cbHandler,
+			}
+			changeList = append(changeList, &item)
+		}
+	}
+
+	return changeList
+}
+
+// createDryrunResource creates the yaml file of given authorization policy spec in a local directory
+func (d *DryRunHandler) createDryrunResource(item *Item) error {
+	convertedCR := item.Resource
+	authzPolicyName := item.Resource.ConfigMeta.Name
+	namespace := item.Resource.ConfigMeta.Namespace
+	convertedObj, err := crd.ConvertConfig(collections.IstioSecurityV1Beta1Authorizationpolicies, convertedCR)
+	if err != nil {
+		return fmt.Errorf("unable to convert authorization policy config to istio objects, resource name: %v", convertedCR.Name)
+	}
+	configInBytes, err := yaml.Marshal(convertedObj)
+	if err != nil {
+		return fmt.Errorf("could not marshal %v: %v", convertedCR.Name, err)
+	}
+	yamlFileName := authzPolicyName + "--" + namespace + ".yaml"
+	return ioutil.WriteFile(dryRunStoredFilesDirectory+yamlFileName, configInBytes, 0644)
+}
+
+// findDeleteDryrunResource retrieves the yaml file from local directory and deletes it
+func (d *DryRunHandler) findDeleteDryrunResource(item *Item) error {
+	authzPolicyName := item.Resource.ConfigMeta.Name
+	namespace := item.Resource.ConfigMeta.Namespace
+	yamlFileName := authzPolicyName + "--" + namespace + ".yaml"
+	if _, err := os.Stat(dryRunStoredFilesDirectory + yamlFileName); os.IsNotExist(err) {
+		log.Infof("file %s does not exist in local directory", dryRunStoredFilesDirectory+yamlFileName)
+		return nil
+	} else if err != nil {
+		log.Infof("error stating file %s in local directory: %s, error: %s", dryRunStoredFilesDirectory+yamlFileName, err)
+	}
+	log.Infof("deleting file under path: %s\n", dryRunStoredFilesDirectory+yamlFileName)
+	return os.Remove(dryRunStoredFilesDirectory + yamlFileName)
+}
+
+// ReadConvertToModelConfig reads in the authorization policy yaml object and converts it into a model.Config struct
+func ReadConvertToModelConfig(serviceName, namespace string) (*model.Config, error) {
```

Comment:
Can we add simple unit tests for the functions createDryrunResource, findDeleteDryrunResource, ReadConvertToModelConfig, and FetchServices? We can use temp dir to interact with files: https://golang.org/pkg/os/#TempDir